sdk/openai/openai-assistants/src/api/assistantMessages/index.ts

```diff
@@ -0,0 +1,344 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import {
+  ListResponseOf,
+  AssistantMessage,
+  AssistantRole,
+  AssistantMessageFile,
+} from "../../models/models.js";
+import {
+  AssistantsContext as Client,
+  CreateMessage200Response,
+  ListMessageFiles200Response,
+  ListMessages200Response,
+  ModifyMessage200Response,
+  RetrieveMessage200Response,
+  RetrieveMessageFile200Response,
+} from "../../rest/index.js";
+import {
+  StreamableMethod,
+  operationOptionsToRequestParameters,
+} from "@azure-rest/core-client";
+import {
+  AssistantMessagesCreateMessageOptions,
+  AssistantMessagesListMessagesOptions,
+  AssistantMessagesRetrieveMessageOptions,
+  AssistantMessagesModifyMessageOptions,
+  AssistantMessagesListMessageFilesOptions,
+  AssistantMessagesRetrieveMessageFileOptions,
+} from "../../models/options.js";
+
+export function _createMessageSend(
+  context: Client,
+  threadId: string,
+  role: AssistantRole,
+  content: string,
+  options: AssistantMessagesCreateMessageOptions = { requestOptions: {} }
+): StreamableMethod<CreateMessage200Response> {
+  return context
+    .path("/threads/{threadId}/messages", threadId)
+    .post({
+      ...operationOptionsToRequestParameters(options),
+      body: {
+        role: role,
+        content: content,
+        file_ids: options?.fileIds,
+        metadata: options?.metadata,
+      },
+    });
+}
+
+export async function _createMessageDeserialize(
+  result: CreateMessage200Response
+): Promise<AssistantMessage> {
+  if (result.status !== "200") {
+    throw result.body;
```

Comment:
We'll need to carry over all the throwing changes to the src folder even to any non-customized code.